### DISCUSSION TITLE
Start and control MIDI file playback from fluidsynth's shell?

### DISCUSSION THREAD

**Discussion by nickbailey** [FluidSynth/fluidsynth#1109] [Q&A] [answered]
_Is is possible to start and control midi file playback at the fluidsynth shell prompt?_

Context: I'm using fluidsynth 2.1.7 on Debian 11 to perform microtonal music and it's working very well. I've written a very small tkinter application to set tunings and programs so the keyboardist doesn't type commands into the fluidsynth shell. It just opens fluidsynth as a writable pipe and sends commands which are received by fluidsynth.

Now the end-user wants to play a MIDI file though the setup, maybe by dropping it on the python app. I'm wondering if I can start and stop such playback by issuing fluidsynth shell commands. I understand I could rewrite the whole thing using the library, but if it comes to that I might as well just ask the user to use a sequencer instead.

**Comment by derselbst** [comment, marked as answer]
Shell commands for the MIDI player have been added in fluidsynth 2.2.0. Execute `help player` for more details.

**Reply by jimhen3ry** [reply]
> help player

player_start               Start playing from the beginning of current song
player_stop                Stop playing (cannot be executed in a user command file)
player_cont                Continue playing (cannot be executed in a user command file)
player_seek num            Move forward/backward in current song to +/-num ticks
player_next                Move to next song (cannot be executed in a user command file)
player_loop num            Set loop number to num (-1 = loop forever)
player_tempo_bpm num       Set tempo to num beats per minute
player_tempo_int [mul]     Set internal tempo multiplied by mul (default mul=1.0)

Is there a way to load a MIDI file to the player from the FluidSynth shell command prompt?

**Reply by derselbst** [reply]
> Is there a way to load a MIDI file to the player from the FluidSynth shell command prompt?

Currently not.

**Reply by nickbailey** [reply]
Thanks for confirming that. Maybe player_clear_playlist and player_add_item (song, no! ;) ) would do the trick.

I'm ok at C and C++. Maybe I should like around the source a bit...
